services/app/src/components/views/Search.tsx

```diff
@@ -98,43 +99,72 @@ function renderNoQuests(props: Props): JSX.Element {
   </Card>);
 }
 
-export function search(props: Props): JSX.Element {
-  if (!props.user.loggedIn) {
-    return renderStoredQuests(props);
-  }
+export class Search extends React.Component<Props, {}> {
 
-  if (props.searching) {
-    return renderLoading(props);
+  public componentDidMount() {
+    if (!this.props.results) {
```

Comment:
AIUI this trips when a user isn't logged in (where we should just be showing stored quests). Does this cause an error? Can we test this?

Reply:
Good thought; just tested, nothing breaks in incognito

Reply:
I mean a unit test... thoughts:
- doesn't trigger a search attempt when user not logged in
- searches when no search results on initial page view
- doesn't search when there are already search results